src/main/java/uk/gov/justice/hmpps/prison/service/OffenderTransactionHistoryService.java

```diff
@@ -2,73 +2,69 @@
 
 @Service
 @Transactional(readOnly = true)
 @Validated
 @Slf4j
 public class OffenderTransactionHistoryService {
 
-    private String apiCurrency;
-    private OffenderTransactionHistoryRepository historyRepository;
-    private OffenderRepository offenderRepository;
+    private final String apiCurrency;
+    private final OffenderTransactionHistoryRepository historyRepository;
 
     public OffenderTransactionHistoryService(@Value("${api.currency:GBP}") final String currency,
-                                             final OffenderTransactionHistoryRepository historyRepository,
-                                             final OffenderRepository offenderRepository) {
+                                             final OffenderTransactionHistoryRepository historyRepository) {
         this.apiCurrency = currency;
         this.historyRepository = historyRepository;
-        this.offenderRepository = offenderRepository;
     }
 
-    public static final Comparator<OffenderTransactionHistory> TRANSACTION_HISTORY_SORTING_POLICY = Comparator
-            .comparing(OffenderTransactionHistory::getEntryDate)
-            .thenComparing(Comparator.comparing(OffenderTransactionHistory::getTransactionEntrySequence).reversed())
-            .reversed();
+    public static final Comparator<OffenderTransactionHistory> SORT_BY_MOST_RECENT_DATE_THEN_BY_LATEST_SEQ = Comparator
+            .comparing(OffenderTransactionHistory::getEntryDate).reversed()
+            .thenComparing(Comparator.comparing(OffenderTransactionHistory::getTransactionEntrySequence).reversed());
 
     @VerifyOffenderAccess
     public List<OffenderTransactionHistoryDto> getTransactionHistory(final String offenderNo,
                                                                      final Optional<String> accountCodeOpl,
                                                                      final Optional<LocalDate> fromDateOpl,
                                                                      final Optional<LocalDate> toDateOpl) {
-       validate(offenderNo, accountCodeOpl, fromDateOpl,toDateOpl);
+        validate(offenderNo, accountCodeOpl, fromDateOpl, toDateOpl);
 
         var fromDate = fromDateOpl.orElse(LocalDate.now());
         var toDate = toDateOpl.orElse(LocalDate.now());
         checkDateRange(fromDate, toDate);
 
-        if(accountCodeOpl.isPresent()) {
+        if (accountCodeOpl.isPresent()) {
             boolean isAccountCodeExists = AccountCode.exists(accountCodeOpl.get());
             checkState(isAccountCodeExists, "Unknown account-code " + accountCodeOpl.get());
         }
 
-        return getSortedHistories(offenderNo, accountCodeOpl, fromDate, toDate).stream()
-                .map(h -> Pair.of(h, apiCurrency))
-                .map(OffenderTransactionHistoryTransformer::transform)
-                .collect(Collectors.toList());
+        return accountCodeOpl.map(AccountCode::byCodeName)
+            .filter(Optional::isPresent)
```

Comment:
Do you need a test for this filtering?

Reply:
These is an existing test that covers both routes. One being when the account code is available and the other when not. I've added a couple of tests for mapping and these also make sure that both paths are tested.